Services/ItemService.cs

```diff
@@ -1,12 +1,66 @@
 ﻿using DomainModels;
+using Repositories;
 using Services.DTO;
 using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace Services
 {
-    public class ItemService
+    public class ItemService:IItemService
     {
+                private readonly List<int> numbers = new List<int> { 1, 2, 3 };
+        public readonly IItemRepository _itemRepository;
+
+        public ItemService( IItemRepository itemRepository)
+        {
+            _itemRepository = itemRepository;
+
+        }
+        public IEnumerable<ItemDTO> GetAll()
+        {
+            var items = _itemRepository.All();
+            IEnumerable<ItemDTO> response = items.Select(s => new ItemDTO() { Id = s.Id, Text = s.Text });
+            return response;
+        }
+
+        public IEnumerable<ItemDTO> GetAllByFilter(ItemByFilterDTO filters)
+        {
+            var items = _itemRepository.All();
+            IEnumerable<ItemDTO> response = items.Select(s => new ItemDTO() { Id = s.Id, Text = s.Text }).Where(x => x.Text == filters.Text);
```

Comment:
A better practice is to perform `Where` before `Select` so that the data is filtered before the conversion to `ItemDTO`. No need to modify just for your info